cmd/soroban-rpc/internal/preflight/preflight.go

```diff
@@ -79,6 +87,7 @@ type PreflightParameters struct {
 	NetworkPassphrase string
 	LedgerEntryReadTx db.LedgerEntryReadTx
 	BucketListSize    uint64
+	ResourceConfig    *ResourceConfig
```

Comment:
I don't think this should be a pointer. If it was ommitted in the request, we can initialize it to the default value.

In fact ... we should probably always use a base default value and add (or substract) the passed value on top of it

Reply:
I think it should be optional in the request, but internally it can just be set to a default like you said. I don't think it should be additive, though: I'd rather we document the default and let users override it.

Reply:
Initialized this to a default value instead of being a pointer here. Since it's a pointer in `SimulateTransactionRequest` it will be optional (and set to default if `nil`)